test/integration/acceptance/query_permission_test_acc_details.cpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <regex>
+
+#include "query_permission_test_acc_details.hpp"
+
+using interface::types::AccountDetailKeyType;
+using interface::types::AccountDetailValueType;
+
+QueryPermissionAccDetails::QueryPermissionAccDetails()
+    : QueryPermissionTestBase({Role::kGetMyAccDetail},
+                              {Role::kGetDomainAccDetail},
+                              {Role::kGetAllAccDetail}) {
+      user_acc_details_.emplace(std::make_pair(AccountDetailKeyType("key1"), AccountDetailValueType("val1")));
+      user_acc_details_.emplace(std::make_pair(AccountDetailKeyType("key2"), AccountDetailValueType("val2")));
+}
+
+IntegrationTestFramework &QueryPermissionAccDetails::prepareState(
+    AcceptanceFixture *fixture,
+    const interface::RolePermissionSet &spectator_permissions /*= {}*/) {
+  auto target_permissions = spectator_permissions;
+  target_permissions.set(Role::kSetDetail);
+
+  // initial state setup
+  auto &itf = QueryPermissionTestBase::prepareState(
+      fixture, spectator_permissions, target_permissions);
+
+  for (const auto& details : user_acc_details_) {
+    itf.sendTxAwait(fixture->complete(fixture->baseTx(kUserId).setAccountDetail(
+                        kUserId, details.first, details.second)),
+                    getBlockTransactionsAmountChecker(1));
+  }
+
+  return itf;
+}
+
+/// check JSON value by key, disregarding hierarchy
+void checkJSONvalue(const std::string &json_str,
+                    const std::string &key,
+                    const std::string &value) {
+  std::regex keyval_regex(
+      std::string("\"") + key + R"("[ \t\r\n]*:[ \t\r\n]*")" + value + "\"",
```

Comment:
`\t\r\n` sequence is per library-defined. JSON formatting is completely optional, thus this regex isn't optional. Probably json lib (e.g our rapidjson) is be a better choice here

Reply:
not sure I got your point. The purpose of this test is to check permissions, not complex JSON values in profile details. This method of checking JSON key-value pairs seems enough while we are limited to string type and do not care of hierarchy.